chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -391,55 +392,45 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
 
     fn check_block_reward_maturity_settings(&self, block: &Block) -> Result<(), CheckBlockError> {
         let required = block.consensus_data().reward_maturity_distance(self.chain_config);
+
         for output in block.block_reward().outputs() {
             match output.purpose() {
-                common::chain::OutputPurpose::Transfer(_) => {
-                    return Err(CheckBlockError::InvalidBlockRewardOutputType(
-                        block.get_id(),
-                    ))
-                }
-                common::chain::OutputPurpose::LockThenTransfer(_, tl) => match tl {
-                    common::chain::timelock::OutputTimeLock::UntilHeight(_) => {
-                        return Err(CheckBlockError::InvalidBlockRewardMaturityTimelockType(
-                            block.get_id(),
-                        ))
-                    }
-                    common::chain::timelock::OutputTimeLock::UntilTime(_) => {
-                        return Err(CheckBlockError::InvalidBlockRewardMaturityTimelockType(
-                            block.get_id(),
-                        ))
-                    }
-                    common::chain::timelock::OutputTimeLock::ForBlockCount(c) => {
-                        let cs: i64 = (*c)
-                            .try_into()
-                            .map_err(|_| {
-                                CheckBlockError::InvalidBlockRewardMaturityDistanceValue(
+                OutputPurpose::LockThenTransfer(_, tl) => {
```

Comment:
You could've still used a separate function... no problem in that. I don't know why you reverted it.

Reply:
Originally I moved the code to a closure because it was called twice, but without it, I don't see the point in separate function it's not that big. If you think it's a good thing to do I will.

Reply:
If there are things around it, I think it's a good thing.